code/game/objects/items/reagent_containers/blood_pack.dm

```diff
@@ -27,6 +32,94 @@
 		if(10 to 50) icon_state = "half"
 		if(51 to INFINITY) icon_state = "full"
 
+/obj/item/reagent_container/blood/attack(mob/attacked_mob, mob/user)
+	. = ..()
+
+	if(connected_to == attacked_mob)
+		connected_to = null
+		STOP_PROCESSING(SSobj, src)
+		user.visible_message("[user] detaches \the [src] from \the [connected_to].", \
+			"You detach \the [src] from \the [connected_to].")
+		return
+
+	if(!skillcheck(user, SKILL_SURGERY, SKILL_SURGERY_NOVICE))
+		return
+
+	if(user.action_busy)
+		return
+
+	if(!do_after(user, 3 SECONDS * user.get_skill_duration_multiplier(SKILL_SURGERY), INTERRUPT_ALL, BUSY_ICON_FRIENDLY, attacked_mob, INTERRUPT_MOVED, BUSY_ICON_MEDICAL))
+		to_chat(user, SPAN_WARNING("You were interrupted before you could finish!"))
+		return
+
+	if(istype(attacked_mob, /mob/living/carbon/human))
+		connected_to = attacked_mob
+		START_PROCESSING(SSobj, src)
+		user.visible_message("[user] attaches [src] to [connected_to].", \
+			"You attach [src] to [connected_to].")
+
+/obj/item/reagent_container/blood/dropped(mob/user)
+	. = ..()
+
+	if(connected_to)
+		connected_to.visible_message("[src] breaks free of [connected_to]!", "[src] is pulled out of you!")
+		connected_to.apply_damage(3, BRUTE, pick("r_arm", "l_arm"))
+		connected_to.emote("scream")
+		connected_to = null
+		STOP_PROCESSING(SSobj, src)
+		return
+
+/obj/item/reagent_container/blood/process()
+	if(!connected_to)
+		STOP_PROCESSING(SSobj, src)
```

Comment:
missing a return here i think?

note that you can also simply `return PROCESS_KILL` in process() instead of using STOP_PROCESSING, which stops scheduling on whatever processing subsystem is running it

Reply:
TIL and correct.

Will change thank you!